typescript/sdk/src/hook/EvmHookModule.hardhat-test.ts

```diff
@@ -350,6 +353,246 @@ describe('EvmHookModule', async () => {
+
+    it('should not update if a domain routing hook includes an address of an existing hook (depth 2)', async () => {
+      const owner = await multiProvider.getSignerAddress(chain);
+      const config: DomainRoutingHookConfig = {
+        type: HookType.ROUTING,
+        owner,
+        domains: {
+          9913371: {
+            type: HookType.AGGREGATION,
+            hooks: [
+              {
+                type: HookType.MERKLE_TREE,
+              },
+              {
+                owner,
+                type: HookType.PROTOCOL_FEE,
+                maxProtocolFee: '1',
+                protocolFee: '0',
+                beneficiary: owner,
+              },
+            ],
+          },
+          9913372: {
+            type: HookType.AGGREGATION,
+            hooks: [
+              {
+                type: HookType.MERKLE_TREE,
+              },
+              {
+                owner,
+                type: HookType.PROTOCOL_FEE,
+                maxProtocolFee: '1',
+                protocolFee: '0',
+                beneficiary: owner,
+              },
+            ],
+          },
+        },
+      };
+      // create a new hook
+      const { hook, initialHookAddress } = await createHook(config);
+
+      // Set the deepest hooks to their addresses
+      const expectedConfig: any = deepCopy(await hook.read());
+      expectedConfig.domains.test1.hooks[0] =
+        expectedConfig.domains.test1.hooks[0].address;
+      expectedConfig.domains.test2.hooks[0] =
+        expectedConfig.domains.test2.hooks[0].address;
+
+      await expectTxsAndUpdate(hook, expectedConfig, 0);
+      expect(eqAddress(initialHookAddress, hook.serialize().deployedHook)).to.be
+        .true;
+    });
+
+    it('should not update if a fallback routing hook includes an address of an existing hook', async () => {
+      const owner = await multiProvider.getSignerAddress(chain);
+      const config: FallbackRoutingHookConfig = {
+        type: HookType.FALLBACK_ROUTING,
+        owner,
+        domains: {
+          9913371: randomHookConfig(0, 2),
+          9913372: randomHookConfig(0, 2),
+        },
+        fallback: randomHookConfig(0, 2),
+      };
+      // create a new hook
+      const { hook, initialHookAddress } = await createHook(config);
+
+      const derivedHook = (await hook.read()) as FallbackRoutingHookConfig;
+      const { test1: firstHook, test2: secondHook } = derivedHook.domains;
+
+      const expectedConfig = {
+        ...config,
+        domains: {
+          test1: (firstHook as DerivedHookConfig).address,
+          test2: secondHook,
+        },
+        fallback: (derivedHook.fallback as DerivedHookConfig).address,
+      };
+
+      await expectTxsAndUpdate(hook, expectedConfig, 0);
+      expect(eqAddress(initialHookAddress, hook.serialize().deployedHook)).to.be
+        .true;
+    });
+
+    it('should not update if a fallback routing hook includes an address of an existing hook (depth 2)', async () => {
```

Comment:
kinda ugly but open to a better way to generate random configs with addresses in random depths for more coverage